### PR TITLE
Hardcoded API for check, moved api functions to Application Presenter

### PR DESCRIPTION
**Pull request by arifahmedozturk**

### CODE REVIEW

File: SharedCode/src/commonMain/kotlin/ApplicationContract.kt [hunk 1]
```diff
@@ -9,5 +9,9 @@ interface ApplicationContract {
 
     abstract class Presenter: CoroutineScope {
         abstract fun onViewTaken(view: View)
+        abstract fun getJourneys():String
+
     }
+
+
```
Comment (kicferk1) [inline SharedCode/src/commonMain/kotlin/ApplicationContract.kt:16]: Probably don't need all the whitespace here :)

File: SharedCode/src/commonMain/kotlin/ApplicationPresenter.kt [hunk 2]
```diff
@@ -15,5 +17,20 @@ class ApplicationPresenter: ApplicationContract.Presenter() {
     override fun onViewTaken(view: ApplicationContract.View) {
         this.view = view
         view.setLabel(createApplicationScreenMessage())
+
+    }
+    private suspend fun getApiCall():String {
+        var content = ""
```
Comment (kicferk1) [inline SharedCode/src/commonMain/kotlin/ApplicationPresenter.kt:23]: As your IDE says, probably don't need to initialise it with empty string, just initialise it with the result of client.get(...)

File: SharedCode/src/commonMain/kotlin/ApplicationPresenter.kt [hunk 3]
```diff
@@ -15,5 +17,20 @@ class ApplicationPresenter: ApplicationContract.Presenter() {
...
+        content = client.get<String>("https://mobile-api-dev.lner.co.uk/v1/fares?originStation=1444&destinationStation=HML&viaStation=WFJ&outboundDateTime=2020-07-20T12%3A16%3A27.371%2B00%3A00&numberOfChildren=2&numberOfAdults=2&doSplitTicketing=false")
+        client.close()
+        return content
+    }
+    override fun getJourneys():String
+    {
+        launch{
+            getApiCall()
+        }
+        return ""
```
Comment (kicferk1) [inline SharedCode/src/commonMain/kotlin/ApplicationPresenter.kt:34]: In another PR we need to actually access the results somehow.

File: SharedCode/src/commonMain/kotlin/ApplicationPresenter.kt [hunk 4]
```diff
@@ -19,18 +19,19 @@ class ApplicationPresenter: ApplicationContract.Presenter() {
         view.setLabel(createApplicationScreenMessage())
 
     }
-    private suspend fun getApiCall():String {
-        var content = ""
+    private suspend fun getApiCall(originStation:String,destinationStation:String):String {
         val client = HttpClient()
-        content = client.get<String>("https://mobile-api-dev.lner.co.uk/v1/fares?originStation=1444&destinationStation=HML&viaStation=WFJ&outboundDateTime=2020-07-20T12%3A16%3A27.371%2B00%3A00&numberOfChildren=2&numberOfAdults=2&doSplitTicketing=false")
+        val queryURL = "https://mobile-api-dev.lner.co.uk/v1/fares?originStation=".plus(originStation).plus("&destinationStation=").plus(destinationStation).plus("&viaStation=WFJ&outboundDateTime=2020-07-20T12%3A16%3A27.371%2B00%3A00&numberOfChildren=2&numberOfAdults=2&doSplitTicketing=false")
```
Comment (kicferk1) [inline SharedCode/src/commonMain/kotlin/ApplicationPresenter.kt:23]: String interpolation is a neat trick, you can do something like 
```
val station = "EXT"
val query = "originStation=$EXT"
```
Which lets you get rid of pluses.

File: SharedCode/src/commonMain/kotlin/ApplicationPresenter.kt [hunk 5]
```diff
@@ -19,18 +19,19 @@ class ApplicationPresenter: ApplicationContract.Presenter() {
...
-    override fun getJourneys():String
+    override fun getJourneys(originStation:String,destinationStation:String):String
     {
+        var result = ""
         launch{
-            getApiCall()
+            result = getApiCall(originStation,destinationStation)
         }
-        return ""
+        return result
```
Comment (kicferk1) [inline SharedCode/src/commonMain/kotlin/ApplicationPresenter.kt:34]: I imagine this will always be "", but that can be fixed in next PR.

Comment (arifahmedozturk) [reply SharedCode/src/commonMain/kotlin/ApplicationPresenter.kt:34]: why? don't I need to get the string to then parse it?

**Pull request merged**